### ISSUE TITLE
TImeout for fetching config

### ISSUE THREAD

**Issue by spinza** [ytti/oxidized#908]
I'm using this to fetch configs from a large non-profit community network.  The devices are not always under my control and some of the devices appear to take a long time to respond.  Particularly some of the devices seem to retain and active SSH connection but take a long time to send the actual config.  

I've seen a device take 45min to fetch a config.  This is fetched via SSH with an SSH timeout set. Which means the SSH timeout is not triggering which guess means the SSH connection is fine?

Can we do a overall config fetching time-out to avoid situations like this.  I.e. if config not fetched inside a 5 min then stop the job and move on?

I suspect this may also tie up workers causing delays in other devices being fetched.

**Comment by ytti** [comment]
When you do it manually when it takes long time, what is happening on the screen? As long as there input within timeout, we should be fine.
We do not have overall timeout for specific node, it wouldn't be hard to implement.

**Comment by spinza** [comment]
Definitely a weird error on the device.  It just hangs at a point but eventually completes with error messages.

**Comment by spinza** [comment]
If you give me a hint on how to do it I'd be happy to add.

**Comment by ytti** [comment]
What is your timeout currently? To me it sounds like there is no output coming from system for long time, that should trigger timeout.

**Comment by spinza** [comment]
15 seconds.

**Comment by ytti** [comment]
It really should give up after 15s and raise timeout. When we run any command via ```cmd```, we expect to see prompt or data in ```timeout``` if we don't see prompt or data, we give up.

So really if there is no input for 15s, it already should be timing out.

Can you test with telnet, if behaviour is same? Can you reliably reproduce it in SSH, or does it only happen occasionally?

**Comment by spinza** [comment]
Not too difficult to repeat now.  I can certainly make it stand still for 15s at least.  Is there a logging option that will help?

**Comment by ytti** [comment]
There is really only:
```
debug: true
input:
  debug: true
```

First adds more verbose debugging, second adds input  (telnet, ssh, ftp...) session logging.

**Comment by spinza** [comment]
I've looked at this further and come to the following conclusions:

The code passes the timeout to the Net::SSH module.  Apparently the timeout there is apparently only a connection timeout.  So, once a connection is made it will not timeout based on that setting.  [Found this here:](https://net-ssh.github.io/net-ssh/classes/Net/SSH.html)
> :timeout => how long to wait for the initial connection to be made

I've managed get a timeout by setting `ServerAliveInterval` and `ServerAliveCountMax` in the oxidized user's ssh config.
